src/lib/pubkey/classic_mceliece/cmce_poly.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Classic McEliece Polynomials
+ * Based on the public domain reference implementation by the designers
+ * (https://classic.mceliece.org/impl.html - released in Oct 2022 for NISTPQC-R4)
+ *
+ * (C) 2023 Jack Lloyd
+ *     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#include <botan/internal/cmce_poly.h>
+#include <botan/internal/ct_utils.h>
+#include <botan/internal/stl_util.h>
+
+namespace Botan {
+
+namespace {
+
+/**
+ * @brief loads @p bytes into a GF_Elem vector in Little Endian.
+ *
+ * Replaces load_le<GF_Elem> as it cannot be compiled on Big Endian architectures
+ *
+ * @param bytes input bytes
+ * @return The vector of GF_Elem elements
+ */
+std::vector<GF_Elem> load_le_gf_vec(std::span<const uint8_t> bytes) {
+   std::vector<GF_Elem> result;
+   BufferSlicer bs(bytes);
+   result.reserve(bytes.size() / sizeof(GF_Elem));
+   for(size_t i = 0; i < bytes.size() / sizeof(GF_Elem); ++i) {
+      result.emplace_back(load_le<uint16_t>(bs.take<sizeof(GF_Elem)>()));
+   }
+   return result;
+}
+
+}  // namespace
+
+Classic_McEliece_GF Classic_McEliece_Polynomial::operator()(Classic_McEliece_GF a) const {
+   BOTAN_ASSERT(a.modulus() == coef_at(0).modulus(), "Unmatching Galois fields");
+
+   Classic_McEliece_GF r(GF_Elem(0), a.modulus());
+   for(auto it = m_coef.rbegin(); it != m_coef.rend(); ++it) {
+      r *= a;
+      r += *it;
+   }
+
+   return r;
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::multiply(const Classic_McEliece_Polynomial& a,
+                                                                       const Classic_McEliece_Polynomial& b) const {
+   std::vector<Classic_McEliece_GF> prod(m_t * 2 - 1, {GF_Elem(0), m_poly_f});
+
+   for(size_t i = 0; i < m_t; ++i) {
+      for(size_t j = 0; j < m_t; ++j) {
+         prod.at(i + j) += (a.coef_at(i) * b.coef_at(j));
+      }
+   }
+
+   for(size_t i = (m_t - 1) * 2; i >= m_t; --i) {
+      for(auto& [idx, coef] : m_position_map) {
+         prod.at(i - m_t + idx) += coef * prod.at(i);
+      }
+   }
+
+   prod.erase(prod.begin() + m_t, prod.end());
+
+   return Classic_McEliece_Polynomial(prod);
+}
+
+Classic_McEliece_Polynomial Classic_McEliece_Polynomial_Ring::create_element_from_bytes(
+   std::span<const uint8_t> bytes) const {
+   BOTAN_ARG_CHECK(bytes.size() == m_t * 2, "Correct input size");
+   auto coef = load_le_gf_vec(bytes);
+   return create_element_from_coef(coef);
```

Comment:
Once https://github.com/randombit/botan/pull/3869 is merged, this could be replaced with:

```suggestion
   return create_element_from_coef(load_le<std::vector<GF_Elem>>(bytes));
```

Obsoleting the helper method `load_le_gf_vec().